### PR TITLE
Maven: Use the Maven defaults when building a project

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Populate all ProjectBuildingRequests with the Maven default values by using
the MavenExecutionRequestPopulator. This better aligns the tool with the
behavior of Maven.

The main reason for this is that it adds the default Maven central
repository to the remote repositories of the request. This is important for
projects that use a parent POM from the central repository because it could
otherwise not be resolved. Usually the remote repository for a parent POM
would have to be defined in the POM itself, the only exception is if the
parent POM comes from Maven central, because this repository is always used
by Maven.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/329)

### CODE REVIEW

File: analyzer/src/funTest/kotlin/MavenTest.kt [hunk 1]
```diff
@@ -101,5 +101,20 @@ class MavenTest : StringSpec() {
 
             yamlMapper.writeValueAsString(result) shouldBe expectedResult
         }
+
+        "Parent POM from Maven central can be resolved" {
+            // Delete the parent POM from the local repository to make sure it has to be resolved from Maven central.
+            val userDir = File(System.getProperty("user.dir"))
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/MavenTest.kt]: This should be `user.home`, see https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html.

File: analyzer/src/funTest/kotlin/MavenTest.kt [hunk 2]
```diff
@@ -101,5 +101,20 @@ class MavenTest : StringSpec() {
...
+            File(userDir, ".m2/repository/org/springframework/boot/spring-boot-starter-parent/1.5.3.RELEASE")
+                    .deleteRecursively()
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/MavenTest.kt]: Better `.safeDeleteRecursively()`?

File: analyzer/src/main/kotlin/MavenSupport.kt [hunk 3]
```diff
@@ -129,14 +130,22 @@ class MavenSupport(localRepositoryManagerConverter: (LocalRepositoryManager) ->
...
 
-    fun createProjectBuildingRequest(resolveDependencies: Boolean) =
-            DefaultProjectBuildingRequest().apply {
-                isResolveDependencies = resolveDependencies
-                repositorySession = repositorySystemSession
-                systemProperties["java.home"] = System.getProperty("java.home")
-                systemProperties["java.version"] = System.getProperty("java.version")
-                validationLevel = ModelBuildingRequest.VALIDATION_LEVEL_MINIMAL
-            }
+    fun createProjectBuildingRequest(resolveDependencies: Boolean) : ProjectBuildingRequest {
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/MavenSupport.kt]: Nit: Wrong space before `:`.

File: analyzer/src/funTest/kotlin/MavenTest.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/MavenTest.kt]: Done.

File: analyzer/src/funTest/kotlin/MavenTest.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/funTest/kotlin/MavenTest.kt]: Done.

File: analyzer/src/main/kotlin/MavenSupport.kt (hunk 3 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/MavenSupport.kt]: Done.

**Pull request merged**